src/common/HTTPRequest.ts

```diff
@@ -572,11 +573,13 @@ export class HTTPRequest {
         Buffer.byteLength(responseBody)
       );
 
+    const status = response.status || 200;
     await this._client
       .send('Fetch.fulfillRequest', {
         requestId: this._interceptionId,
-        responseCode: response.status || 200,
-        responsePhrase: STATUS_TEXTS[response.status || 200],
+        responseCode: status,
+        responsePhrase:
+          STATUS_TEXTS[String(status) as keyof typeof STATUS_TEXTS],
```

Comment:
How are we sure that this assertion is valid? Can you add a comment to describe it?

Also, I wonder if we can type `ResponseForRequest` more strictly for the values that `status` can get?
If we're sure that it will be one of the valid responses from the `STATUS_TEXTS`; maybe we can
* Change the key of `STATUS_TEXTS` to be number
* Update `ResponseForRequest`s `status` field to be `keyof typeof STATUS_TEXTS`
* Remove the `String(status)` conversion here

AFAIU that change would also mean a breaking change though since the users won't be able to provide a `ResponseForRequest` with a `status` different than the ones in `STATUS_TEXTS`. WDYT?

Reply:
Good point, I think we can just change the index signature of STATUS_TEXTS here!

Reply:
Hihiiim I feel like typing `STATUS_TEXTS` as `{ [key: string]: string }` can be erroneous since what I understand from this signature is that, whatever `key` I provide as `string`, I'll get a string back.

However, `STATUS_TEXTS` doesn't ensure this since we're able to get `undefined` value with something like `104`.

So, if we're sure that `response.status` will be one of the keys of `STATUS_TEXTS`, it would be better to type it more strictly (like `keyof typeof STATUS_TEXTS`) or if we don't want to change the typing of `response`, we can keep the assertion and add a comment on why we're sure that `response.status` will be one of the keys of `STATUS_TEXTS`. WDYT?